patches/rosconsole.patch

```diff
@@ -0,0 +1,180 @@
+--- catkin_ws/src/ros_comm/rosconsole/CMakeLists.txt
++++ catkin_ws/src/ros_comm/rosconsole/CMakeLists.txt
+@@ -10,9 +10,10 @@ find_package(catkin REQUIRED COMPONENTS cpp_common rostime rosunit)
+ find_package(Boost COMPONENTS regex system thread)
+ 
+ # select rosconsole backend
+-set(ROSCONSOLE_BACKEND "" CACHE STRING "Type of rosconsole backend, one of 'log4cxx', 'glog', 'print'")
++set(ROSCONSOLE_BACKEND "" CACHE STRING "Type of rosconsole backend, one of 'android', 'log4cxx', 'glog', 'print'")
+ set(rosconsole_backend_INCLUDE_DIRS)
+ set(rosconsole_backend_LIBRARIES)
++
+ if(ROSCONSOLE_BACKEND STREQUAL "" OR ROSCONSOLE_BACKEND STREQUAL "log4cxx")
+   find_package(Log4cxx QUIET)
+   if(NOT LOG4CXX_LIBRARIES)
+@@ -27,6 +28,12 @@ if(ROSCONSOLE_BACKEND STREQUAL "" OR ROSCONSOLE_BACKEND STREQUAL "log4cxx")
+     message(FATAL_ERROR "Couldn't find log4cxx library")
+   endif()
+ endif()
++if(ROSCONSOLE_BACKEND STREQUAL "" OR ROSCONSOLE_BACKEND STREQUAL "android")
++  if(ANDROID)
++    list(APPEND rosconsole_backend_LIBRARIES rosconsole_android rosconsole_backend_interface log)
++    set(ROSCONSOLE_BACKEND "android")
++  endif()
++endif()
+ if(ROSCONSOLE_BACKEND STREQUAL "" OR ROSCONSOLE_BACKEND STREQUAL "glog")
+   find_package(PkgConfig)
+   pkg_check_modules(GLOG QUIET libglog)
+@@ -85,6 +92,9 @@ elseif(ROSCONSOLE_BACKEND STREQUAL "glog")
+ elseif(ROSCONSOLE_BACKEND STREQUAL "print")
+   add_library(rosconsole_print src/rosconsole/impl/rosconsole_print.cpp)
+   target_link_libraries(rosconsole_print rosconsole_backend_interface)
++elseif(ROSCONSOLE_BACKEND STREQUAL "android")
++  add_library(rosconsole_android src/rosconsole/impl/rosconsole_android.cpp)
++  target_link_libraries(rosconsole_android rosconsole_backend_interface log)
+ else()
+   message(FATAL_ERROR "Unknown rosconsole backend '${ROSCONSOLE_BACKEND}'")
+ endif()
+
+--- /dev/null
++++ catkin_ws/src/ros_comm/rosconsole/src/rosconsole/impl/rosconsole_android.cpp
+@@ -0,0 +1,128 @@
++#include "ros/console.h"
++
++#include <android/log.h>
++
++namespace ros
++{
++namespace console
++{
++namespace impl
++{
++
++std::vector<std::pair<std::string, levels::Level> > rosconsole_android_log_levels;
```

Comment:
Is this vector protected against concurrent access?

I do not see any mutex protection in [ros::console::get_loggers()](https://github.com/ros/ros_comm/blob/kinetic-devel/tools/rosconsole/src/rosconsole/rosconsole.cpp#L696) or [ros::console::set_logger_level()](https://github.com/ros/ros_comm/blob/kinetic-devel/tools/rosconsole/src/rosconsole/rosconsole.cpp#L701).